bot/cogs/help.py

```diff
@@ -0,0 +1,236 @@
+                Help syntax : `<Required arguments`.
+                `[t.Optional arguments]`
+                {command.help}
+                """
+            ),
+            color=discord.Color.blue(),
+        )
+        if command.aliases:
+            embed.add_field(name="Aliases :", value="\n".join(command.aliases))
+        embed.set_author(
+            name=str(ctx.message.author),
+            icon_url=str(ctx.message.author.avatar_url),
+        )
+        embed.set_thumbnail(url=str(ctx.bot.user.avatar_url))
+        if command.hidden:
+            embed.set_footer(
+                text=f"Wow, you found {command.name}!",
+                icon_url=str(ctx.bot.user.avatar_url),
+            )
+        else:
+            embed.set_footer(
+                text=f"Are you interested in {command.name}?",
+                icon_url=str(ctx.bot.user.avatar_url),
+            )
+        await ctx.send(embed=embed)
+
+    async def send_group_help(self, group: commands.Group) -> None:
+        """Send help for a group."""
+        ctx = self.context
+        prefix = ctx.prefix
+        embed = discord.Embed(
+            title=textwrap.dedent(
+                f"""
+                Help for group {prefix}
+                {self.get_command_signature(group)}
+                """
+            ),
+            description=textwrap.dedent(
+                f"""
+                Help syntax : `<Required arguments>`.
+                `[t.Optional arguments]`
+                {group.help}
+                """
+            ),
+            color=discord.Color.blue(),
+        )
+        for command in await self.filter_commands(group.commands, sort=True):
+            embed.add_field(
+                name=f"{prefix}{self.get_command_signature(command)}",
+                value=command.help,
+                inline=False,
+            )
+        embed.set_author(
+            name=str(ctx.message.author),
+            icon_url=str(ctx.message.author.avatar_url),
+        )
+        embed.set_thumbnail(url=str(ctx.bot.user.avatar_url))
+        if group.hidden:
+            embed.set_footer(
+                text=f"Wow, you found {group.name}!",
+                icon_url=str(ctx.bot.user.avatar_url),
+            )
+        else:
+            embed.set_footer(
+                text=f"Are you interested in {group.name}?",
+                icon_url=str(ctx.bot.user.avatar_url),
+            )
+        await ctx.send(embed=embed)
+
+    async def send_error_message(self, error: str) -> None:
+        """Send an error message."""
+        ctx = self.context
+        await ctx.bot.httpcat(ctx, 404, error)
+
+
+def setup(bot: commands.Bot) -> None:
+    """Add the help command."""
+    bot.old_help_command = bot.help_command
+    bot.help_command = Help(
+        verify_checks=False,
+        command_attrs={'hidden': True},
+    )
+
+
+def teardown(bot: commands.Bot) -> None:
+    """Remove the help command."""
+    bot.help_command = bot.old_help_command
```

Comment:
```suggestion
def teardown(bot: commands.Bot) -> None:
    """Remove the help command."""
    bot.remove_command("help")
```
not sure what the point of this function is here, but a faster way to remove the help command is to call `bot.remove_command("help")` on initialisatioin

Reply:
That would leave the bot with NO help command.
This behaviour leaves the bot with the default help command

Reply:
not if you were to define help as you would a normal command.

Reply:
That's not how you define help

Reply:
help is defined using `HelpCommand`

Reply:
While I agree with defining help as `HelpCommand` as you did, you could add a `Help` Cog here.

This could actually set the help command to this new help as you're doing in `teardown` but it would now be done in `__init__`, that way we could also use a custom `cog_unload` function which would reset the help back as it was, this could be useful if we needed to disable the help for whatever reason and would provide a more flexible system

Reply:
The teardown function is called when the extension is unloaded.
So you can already disable the help by unloading the extension.
It's added to the bot in setup

Reply:
Yes, but in the way, this is done won't it remove help command completely? Also, we have a way of unloading cogs directly, with `>>unload` which can't be used on this since it's not in a Cog, that's why it would be better to make a separate `Help` cog which controls this instead.

Reply:
In setup, the old help (default one) is stored in a botvar.
On teardown (when it's unloaded), the old help is implemented back from the botvar.
So the help is removed completely

Reply:
Alright, that makes sense, but it might still make more sense to make a cog and store the old help command as a property of that class instead of having it directly saved as a property of `bot`.

Also, can this be unloaded with `>>unload`? Since this is not a cog I don't think that will work there (I might be wrong though) in which case I insist on making this a Cog since we should have the ability to properly unload it.

Reply:
But it is unloaded properly.
And making it a cog doesn't make sense since the HelpCommand isn't a cog.
It doesn't fit in a cog.
And `bot.help_command = bot.help_command.old_help_command` feels wrong
And yes for unload : unload calls `teardown`

Reply:
It's the opposite of `setup` : one is called when you load, the other when you unload

Reply:
Just to make this clear, I'm not asking you to convert this into cog, only to make a cog around it which will handle the loading instead, an example of it:
```py
class Help(Cog):
    def __init__(self, bot):
        self.bot = bot
        self.old_help_command = bot.help_command
         bot.help_command = HelpCommand(
             verify_checks=False,
            command_attrs={'hidden': True},
         )

    def cog_unload(self):
        bot.help_command = self.old_help_command
```
I never meant to actually turn the whole `Help` into a cog, just the load/unload.

Reply:
I got two reasons not to make it a cog :
1) It's completely useless : this cog would be nothing more than a placeholder
2) cogs are a way to group commands, but this cog wouldn't include any command

And where should I put it except in the cogs folder ? We don't have any other extensions folder